### PR TITLE
Pill 07: readable code form, fix to shell variable quoting

### PR DESCRIPTION
**Pull request by Anton-Latukha**
* [x] Now sets are readable and can be more easily understood.
* [x] Converted from DocBook to txt, because found no advantage for DocBook there.
* [x] Since warnings in `nix-repl` would popup - included that warnings, and explained that they are not relevant at that moment.
* [x] Fix shell variable quoting.

### CODE REVIEW

File: pills/07/c-program-derivation.xml [hunk 1]
```diff
@@ -1,7 +0,0 @@
-<screen xmlns="http://docbook.org/ns/docbook"><prompt>nix-repl> </prompt><userinput>:l &lt;nixpkgs></userinput>
-<prompt>nix-repl> </prompt><userinput>simple = derivation { name = "simple"; builder = "${bash}/bin/bash"; args = [ ./simple_builder.sh ]; gcc = gcc; coreutils = coreutils; src = ./simple.c; system = builtins.currentSystem; }</userinput>
```
Comment (jtojnar) [inline pills/07/c-program-derivation.xml]: I would just change this input to multi-line – nix-repl supports line continuation after all.

**Comment by jtojnar** [comment]
It is weird that `div.docbook code` has `white-space: normal` even within `pre`, resulting in

image

instead of

image

(source: https://nixos.org/nixos/nix-pills/working-derivation.html)

**Comment by Anton-Latukha** [comment]
At this point I not found a better way for DocBook then to spam `<sbr/>`.

And this is the best professional suggestion found so far:

"
**Chapter 27. Program listings**

---
You can turn off interpretation of characters that would be recognized as XML markup by enclosing the text in a CDATA section. *(aka "disable the DocBook parsing for code blocks")*

OR

*Formatting listings*

Formatting a program listing for HTML output is best handled by CSS. *(aka "avoid DocBook formating and use direct HTML/CSS for code blocks")*
"
Source: http://www.sagehill.net/docbookxsl/ProgramListings.html

**Comment by jtojnar** [comment]
TXT definitely works much worse than DocBook – it does not format the screen dump, nor does it distinguish between the user input and output.

The line breaks are there in the DocBook, you just need to fix the CSS. Something like the following should work:

```css
div.docbook pre code {
    white-space: pre;
}
```

**Comment by Anton-Latukha** [comment]
Ok. Thank you. I really do not know CSS.

I would do my best, when I get back to this one.

**Comment by grahamc** [comment]
I don't think so, the existing code is well structured and semantically nice. New submissions don't have to be so complicated, but there is not a good reason to undo it for the existing stuff. I'd be interested in PR to fix the whitespace handling.

**Pull request closed without merging**